[Spree Upgrade] Finish up deleting account invoices

luisramos0 opened the pull request:
#### What? Why?
Closes #3533

Additionally this PR:
- cleans up monthly bill description that was left from #3321
- drops account invoices and billable periods tables, for some reason not dropped in #3321

#### What should we test?
Tests are required for the "monthly bill description" removal.
We need to test all the pages where the package selection appears:
- the welcome page that enterprises see after registration
- the drop down menu "change package" in the single enterprise dashboard
- the non-admin list of enterprises

mkllnk commented on db/migrate/20190315225354_delete_all_pending_delayed_jobs.rb:
```diff
@@ -0,0 +1,5 @@
+class DeleteAllPendingDelayedJobs < ActiveRecord::Migration
+  def up
+    Delayed::Job.delete_all
```
I'm not sure this is a good idea. There may be pending jobs that are still important. I think we should delete specific job types that are now obsolete.

luisramos0 replied:
ok, I have changed it to only delete obsolete jobs related to account invoices. Please have a look.
ping @kristinalim are you ok with this?

kristinalim replied:
@luisramos0 Yes, I'm okay with this.

kristinalim replied:
In relation to this comment, I did quick check in recent commits and spotted that `SubscriptionConfirmJob` and `SubscriptionPlacementJob` have been renamed. Pending jobs of this type would not have a corresponding job class in v2. I created #3643 to address this.

luisramos0 replied:
thank you!

mkllnk commented:
We can also keep the pricing information for now and just remove the logic to bill based on that.

luisramos0 commented:
@mkllnk that is all gone already:
- https://github.com/openfoodfoundation/openfoodnetwork/pull/3321/files#diff-59c655beec205769b9b08f9358ca527cL113
- https://github.com/openfoodfoundation/openfoodnetwork/pull/3321/files#diff-7b755a8572efec7fc38da020f5403136
- https://github.com/openfoodfoundation/openfoodnetwork/pull/3321/files#diff-346e21da75b3b2c24c5e7f2a53816ad3

For the placeholder for pricing, we already have 
js.admin.panels.enterprise_package.hub_profile_text1
and 
js.admin.panels.enterprise_package.hub_profile_text1
for each profile, there's plenty of space to write whatever the instance wants about pricing. I dont see why we would need a price specific textarea.

mkllnk commented:
Ah, great. :+1:

What do you think about the delayed jobs?

luisramos0 commented:
ok, so I rebased to resolved conflicts and changed the delayedjobs migration to just delete the obsolete ones: 804375d1aa785e3c4c1c0244b7c4944a3afef255

luisramos0 commented:
ehehe, yes, I thought that must be a shorter way to write this :-D

kristinalim commented on app/views/admin/enterprises/_change_type_form.html.haml:
```diff
@@ -36,9 +33,6 @@
             .top
```
Unless putting back `monthly-pricing-description`, the `top` CSS class might no longer be necessary here.

luisramos0 replied:
you should be right, I'll have a look at this

luisramos0 commented:
@kristinalim this is a dead feature. this specific dynamic part is not configured even in Australia...
we are just deleting it here.
we may get to it later if prioritized but for now I'd not spend more time here.

kristinalim commented:
I'm not sure I understand correctly. Was the point of @mkllnk's concern that some instances bill enterprises offline, and now they won't see the price? Or is that data unused too?

kristinalim commented:
> this specific dynamic part is not configured even in Australia...

So there is no pricing description visible in AU?

luisramos0 commented:
no, pricing is not used anywhere, from live:

FR
![Screenshot 2019-03-22 at 21 09 38](https://user-images.githubusercontent.com/1640378/54853275-49279d00-4ce7-11e9-8df1-5ff827208548.png)
ES
![Screenshot 2019-03-22 at 21 10 29](https://user-images.githubusercontent.com/1640378/54853280-49c03380-4ce7-11e9-99eb-c60f7957fdb3.png)
AU
![Screenshot 2019-03-22 at 21 10 16](https://user-images.githubusercontent.com/1640378/54853281-49c03380-4ce7-11e9-90d0-bbd37001e82a.png)
UK
![Screenshot 2019-03-22 at 21 10 38](https://user-images.githubusercontent.com/1640378/54853278-49c03380-4ce7-11e9-87b4-7580f35ee5a5.png)

this last one is not dynamic is a translation in enGB:
always_free: Plans starting from £1/month

thats why I think we should just delete, it's all been dead for 3 years and can be relived easily if required.
the other option is really to make this monthlyBillingDescription work without the dynamic part, I don't think we should spend time doing that.

luisramos0 commented:
ok, thanks @kristinalim 
I was planning to look at the css but I forgot about it and now that you approved the PR... I will leave it 😊 
moving to test ready

RachL commented:
This is good to go: https://docs.google.com/document/d/1zsKXh8kkQl-JPbUqz-cf8-WfHrjEV2B9a2ocd59NGlM/edit#
